usr/lib/ica_s390_stdll/ica_specific.c

```diff
@@ -221,9 +222,13 @@ static CK_RV load_libica(void)
     void *ibmca_dso = NULL;
 
     /* Load libica */
-    ibmca_dso = dlopen(LIBICA_SHARED_LIB, RTLD_NOW);
+    ibmca_dso = dlopen(LIBICA_SHARED_LIB_V4, RTLD_NOW);
+    if (ibmca_dso == NULL)
+        ibmca_dso = dlopen(LIBICA_SHARED_LIB_V3, RTLD_NOW);
+
     if (ibmca_dso == NULL) {
-        TRACE_ERROR("%s: dlopen(%s) failed\n", __func__, LIBICA_SHARED_LIB);
+        TRACE_ERROR("%s: dlopen(%s or %s) failed\n", __func__,
+                    LIBICA_SHARED_LIB_V4, LIBICA_SHARED_LIB_V3);
```

Comment:
Looks good. However, when we are here, both, loading libica 3 and 4 failed, so should we write: 
`TRACE_ERROR("%s: dlopen(%s and %s) failed\n", __func__,`   ?

Reply:
Well, 'and' sounds like we need to be able to load both.... But one of it is enough, that's why I chose 'or'.